bbq/compiler/desugar.go

```diff
@@ -71,10 +72,11 @@ func NewDesugar(
 		checker:                      checker,
 		importsSet:                   map[common.Location]struct{}{},
 		inheritedFuncsWithConditions: map[string][]*inheritedFunction{},
+		postConditionIndices:         map[ast.Declaration]int{},
 	}
 }
 
-func (d *Desugar) Run() *ast.Program {
+func (d *Desugar) Run() (*ast.Program, map[ast.Declaration]int) {
```

Comment:
Maybe name the return values, especially for the second return value it is not clear what it is just based off of the type